### PR TITLE
AAD-799 notification counter

### PR DESCRIPTION
**Pull request by CathyGroom**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeApp.Pwa.UnitTests/ViewComponents/NotificationCounterTests.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+            taskReminders[0].ReminderStatus = ReminderStatus.Sent;
+
+            var httpContext = new DefaultHttpContext();
+            var apprenticeIdClaim = new Claim(Constants.ApprenticeIdClaimKey, apprenticeId.ToString());
+            var claimsPrincipal = new ClaimsPrincipal(new[] {new ClaimsIdentity(new[]
+        {
+                apprenticeIdClaim
+        })});
+            httpContext.User = claimsPrincipal;
+
```
Comment (alireid) [inline src/SFA.DAS.ApprenticeApp.Pwa.UnitTests/ViewComponents/NotificationCounterTests.cs:40]: whitespace

File: src/SFA.DAS.ApprenticeApp.Pwa.UnitTests/ViewComponents/NotificationCounterTests.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            Assert.That(result, Is.InstanceOf<ViewViewComponentResult>());
+        }
+
+        [Test, MoqAutoData]
+        public async Task Invoke_Async_Returns_Null_Invalid_Apprentice(
+            [Frozen] List<ApprenticeTaskReminder> taskReminders,
+            [Frozen] Mock<IOuterApiClient> client,
+            [Greedy] NotificationCounter viewComponent)
+        {
+            //Arrane
```
Comment (alireid) [inline src/SFA.DAS.ApprenticeApp.Pwa.UnitTests/ViewComponents/NotificationCounterTests.cs:65]: misspelt comment

File: src/SFA.DAS.ApprenticeApp.Pwa/ViewComponents/NotificationCounter.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        public async Task<IViewComponentResult> InvokeAsync()
+        {
+            var apprenticeId = Claims.GetClaim(HttpContext, Constants.ApprenticeIdClaimKey);
+            if(!string.IsNullOrEmpty(apprenticeId))
+            {
+                var notificationsResult = await _client.GetTaskReminderNotifications(new Guid(apprenticeId));
+                if (notificationsResult != null && notificationsResult.TaskReminders.Count > 0)
+                {
+                    int notificationValue = notificationsResult.TaskReminders.Count;
+                    return View("../_NotificationCount", notificationValue);
```
Comment (alireid) [inline src/SFA.DAS.ApprenticeApp.Pwa/ViewComponents/NotificationCounter.cs:25]: Not sure using ../ is safe.. can we use tilde ~ to denote the root and work from that. Safer.

Comment (alireid) [inline src/SFA.DAS.ApprenticeApp.Pwa/ViewComponents/NotificationCounter.cs:25]: Same here with ../ can we look into using tilde to denote root. Paths _may_ behave differently on other platforms other than local.

**Review by alireid — CHANGES_REQUESTED** [review]
Review complete, few tweaks. I'm interested to know how the actual badges are spawned, are we not using Navigator.setAppBadge() etc?
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/setAppBadge

File: src/SFA.DAS.ApprenticeApp.Pwa/ViewComponents/NotificationCounter.cs (hunk 3 above)
Comment (CathyGroom) [reply src/SFA.DAS.ApprenticeApp.Pwa/ViewComponents/NotificationCounter.cs:25]: Same file as before

**Comment by CathyGroom** [comment]
Re the badges - this is currently a tactical solution for short term before we address notifications in more detail.

**Comment by alireid** [comment]
> Re the badges - this is currently a tactical solution for short term before we address notifications in more detail.

Ok understood

**Pull request merged**